boot/nxboot/nxboot_main.c

```diff
@@ -32,24 +32,146 @@
 #include <nxboot.h>
 #include <sys/boardctl.h>
 
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+bool progress_dots_started = false;
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_NXBOOT_PRINTF_PROGRESS
+static const char *g_progress_txt[] =
+{
+  /* Text that will be printed  ...from...    enum progress_msg_e  */
+
+  "*** nxboot ***",                          /* startup_msg */
+  "Power Reset detected, check images only", /* power_reset */
+  "Soft reset detected, check for update",   /* soft_reset */
+  "Found bootable image, boot from primary", /* found_bootable_image */
+  "No bootable image found",                 /* no_bootable_image */
+  "Board failed to boot bootable image",     /* boardioc_image_boot_fail */
+  "Copying bootable image to RAM",           /* ramcopy_started */
+  "Reverting image to recovery",             /* recovery_revert */
+  "Creating recovery image",                 /* recovery_create */
+  "Updating from update image",              /* update_from_update */
+  "Validating primary image",                /* validate_primary */
+  "Validating recovery image",               /* validate_recovery */
+  "Validating update image",                 /* validate_update */
+  "Recovery image created",                  /* recovery_created */
+  "Recovery image invalid, update stopped",  /* recovery_invalid */
+  "Update failed",                           /* update_failed */
```

Comment:
I think that it would be better to associate fields with enum directly with `[ENUM] = "str",` syntax.

Reply:
You're right...it sort of evolved and I didn't go back round the loop again!